src/Google/GlobalSiteTag.php

```diff
@@ -140,6 +140,11 @@ public function register(): void {
 			'update_option_' . $gtag_option_name,
 			function ( $old_value, $value ) use ( $ads_conversion_id ) {
 				if ( $value ) {
+					// Allow the user option value to be updated but do not enable GTG if feature is disabled.
+					if ( ! $this->features->is_enabled( Features::GOOGLE_TAG_GATEWAY ) ) {
```

Comment:
I like the intent of this, but am worried that this could lead to a situation where someone updates the option value when the feature is disabled, and then when the feature is later enabled, the Adapter::update() method is never called.

I think it would be better to always call the update method when the value changes and consider updating the UI in the settings screen to handle the situation where the feature is disabled.

Reply:
Thanks @joemcgill I've reverted this change and we have a new ticket to hide the UI when the GTG feature is disabled via WCS